neps/nep-0393.md

```diff
@@ -104,19 +180,58 @@ SBT registry can define it's own mechanism to atomically recover all tokens rela
 
 SBT Registry based recovery is not part of this specification.
 
-#### Minting authorization
+#### Soul Transfer
+
+The basic use case is described above. Registry MUST provide a permissionless method to allow any user to execute soul transfer. It is essential part of the standard, but the exact interface of the method is not part of the standard, because registries may adopt different mechanism and require different arguments.
+
+Soul transfers must be _semi atomic_. That is, the holder account must be non operational (in terms of SBT supply) until the soul transfer is completed. Given the nature of NEAR blockchain, where transactions are limited by gas, big registries may require to implement the Soul Transfer operation in stages. Source and destination accounts should act as a non-soul accounts while the soul transfer operation is ongoing. For example, in the first call, contract can lock the account, and do maximum X amount of transfers. If the list of to be transferred SBTs has not been exhausted, the contract should keep locking the account and remember the last transferred SBT. Subsequent calls by the same user will resume the operation until the list is exhausted.
+Soul Transfer must emit the `SoulTransfer` event.
 
-A registry can limit which contracts can mint SBTs by implementing a custom issuer registration methods. Example: a simple access control list managed by a DAO.
+Example: Alice has two accounts: `alice1` and `alice2` which she used to mint tokens (see [mint diagram](#minting)). She decides to merge the accounts by doing Soul Transfer.
 
-#### Burning tokens
+```mermaid
+sequenceDiagram
+    actor Alice
+    participant SBT_Registry
+
+    Alice->>SBT_Registry: sbt_soul_transfer(alice1) --accountId alice2
+
+    Alice-->>+SBT_Registry: sbt_tokens_by_owner(alice2)
+    SBT_Registry-->>-Alice: []
+
+    Alice-->>+SBT_Registry: sbt_tokens_by_owner(alice1)
+    SBT_Registry-->>-Alice: [[SBT_1_Contract, [238]], [SBT_2_Contract, [7991, 7992]]]}
+```
+
+### Renewal
+
+Soulbound tokens can have an _expire date_. This is useful for tokens which are related to real world certificates with expire time, or social mechanisms (e.g. community membership). Such tokens SHOULD have an option to be renewable. Examples include mandatory renewal with a frequency to check that the owner is still alive, or renew membership to a DAO that uses SBTs as membership gating.
```

Comment:
"It is useful for tokens which are related to the real-world certificates with an expiration time...."